src/Repositories/DatasetsRepository.php

```diff
@@ -132,23 +135,53 @@ public static function resolve(string $description, array $dataset): array|null
         return $namedData;
     }
 
+    /**
+     * @return Closure|iterable<int|string, mixed>
+     */
+    private static function getScopedDataset(string $name, string $currentTestFile)
+    {
+        $matchingDatasets = array_filter(self::$datasets, function (string $key) use ($name, $currentTestFile) {
+            [$datasetScope, $datasetName] = explode('>>>', $key);
+
+            if ($name !== $datasetName) {
+                return false;
+            }
+
+            if (! str_starts_with($currentTestFile, $datasetScope)) {
+                return false;
+            }
+
+            return true;
+        }, ARRAY_FILTER_USE_KEY);
+
+        $closestScopeDatasetKey = array_reduce(array_keys($matchingDatasets), function ($keyA, $keyB) {
+            if ($keyA === null) {
+                return $keyB;
+            }
+
+            return strlen($keyA) > strlen($keyB) ? $keyA : $keyB;
+        });
+
+        if ($closestScopeDatasetKey === null) {
+            throw new DatasetDoesNotExist($name);
+        }
+
+        return $matchingDatasets[$closestScopeDatasetKey];
+    }
+
     /**
      * @param  array<Closure|iterable<int|string, mixed>|string>  $datasets
      * @return array<array<mixed>>
      */
-    private static function processDatasets(array $datasets): array
+    private static function processDatasets(array $datasets, string $currentTestFile): array
     {
         $processedDatasets = [];
 
         foreach ($datasets as $index => $data) {
             $processedDataset = [];
 
             if (is_string($data)) {
-                if (! array_key_exists($data, self::$datasets)) {
-                    throw new DatasetDoesNotExist($data);
-                }
-
-                $datasets[$index] = self::$datasets[$data];
+                $datasets[$index] = self::getScopedDataset($data, $currentTestFile);
```

Comment:
Small thing, but can the `getScopedDataset` method go below this method?

Reply:
done